### PR TITLE
refactor(source): decouple GitHub polling from Projects, drop author/creator trust checks

### LINKED ISSUE #93: Decouple GitHub source polling from Projects; query issues by label directly
## Problem

`.factory/sources/github` queries GitHub Projects v2 via hand-rolled GraphQL (`.factory/sources/github:60-79`) to match a trigger's `state` against a Project's custom Status field. This exists to answer "which issues are eligible," but eligibility doesn't require a board at all: `gh issue list --label <label> --state open --json ...` answers the same question directly against the issue itself, with no GraphQL, no pagination code we own, and no Projects API dependency.

Board membership is a separate, human-facing concern. An issue a person adds to a project board stays on that board regardless of what Factory reads to decide eligibility — Factory reading labels instead of a Project Status field does not remove anything from anyone's board.

The Jira adapter (`.factory/sources/jira`) already avoids this problem: its JQL is `project + status + labels + creator`, with no board/project concept involved. GitHub is the one adapter still carrying board coupling.

## Trust model (decided, not in scope to re-litigate here)

#91 proposed enforcing an author allowlist in Factory core. Closed: these adapters target private repos/projects where only trusted people have write/triage access at all, so applying a label already functions as the trust boundary. Both adapters drop author/creator-based trust checking entirely rather than reimplement it:

- GitHub: no `--trusted-user` filtering, no `author` matching.
- Jira: drop `creator = currentUser()` from the JQL (`.factory/sources/jira:40`) — it was never a real allowlist and breaks the moment a second team member's tickets need to flow through the trigger.

This depends on an explicit assumption: Factory should not be pointed at a repo/project where untrusted people have label/triage access. Document this assumption directly (README and/or a comment in `.factory/config.toml`) rather than leaving it implicit.

## Expected behavior

`.factory/sources/github` determines eligibility purely from the issue's own `state` (open/closed) and labels, with no call to the Projects API and no author/trust filtering. Pipeline stage (triage vs. implement, etc.) is expressed as which label is present, the same way `.factory/workflows/pr-review/WORKFLOW.md` already manages a `ready-to-merge` label. A human may still add any of these issues to a project board for their own visualization; Factory does not read or depend on that board.

## Acceptance criteria

- `.factory/sources/github` no longer calls `gh api graphql` or references Projects; it uses `gh issue list --state open --label <label> --json number,title,url,labels` (no `author` needed), with `jq` doing pure field reshaping, no `select()`/filtering logic.
- `--project-owner`, `--project-number`, `--status-field`, and `--trusted-user` arguments are removed from the script and from `.factory/config.toml`.
- `.factory/sources/jira` drops `creator = currentUser()` from its JQL; the JQL becomes `project + status + labels` only.
- `state` in each `[trigger.*]` config maps to the issue's native open/closed state; the pipeline stage a trigger cares about is expressed entirely through its `labels`.
- `.factory/workflows/triage/WORKFLOW.md` and `.factory/workflows/implement/WORKFLOW.md` are updated to swap labels at each stage transition (e.g. remove `factory:ready-for-spec`, add `factory:ready-to-implement`) the same way `pr-review` already manages `ready-to-merge`.
- README's GitHub-adapter section is updated to drop the Projects/Status-field explanation and describe the label-only model; no mention of a required board.
- README and/or `.factory/config.toml` documents the trust assumption above (label/triage access is the trust boundary; do not point Factory at a repo/project where that access is not restricted to trusted people).
- Any future Linear adapter is expected to follow the same pattern (plain label/state query, no board dependency, no author/creator filtering) — note this in the adapter-contract docs so none of this is reintroduced.

## Checks

- Manually verify an issue labelled for a trigger is picked up with no Project/board present.
- Manually verify an issue already on an existing project board is unaffected (still visible on the board) after this change.
- Run `factory validate` and `factory workflows` against the updated config.
- Run `cargo test --all-targets`.
- Run `cargo clippy --all-targets --all-features -- -D warnings`.

## Related

- #91 (closed — superseded by the "drop trust checking" decision above)

### PR DESCRIPTION
**Pull request by owainlewis**
## Summary

Implements #93.

- `.factory/sources/github` no longer queries GitHub Projects v2 via hand-rolled GraphQL. It lists issues by native state (open/closed) and labels directly with `gh issue list`, with no Projects API dependency and no board coupling.
- Both `.factory/sources/github` and `.factory/sources/jira` drop author/creator trust filtering entirely. `--trusted-user` matching is gone from the GitHub adapter; `creator = currentUser()` is gone from the Jira adapter's JQL (it was never a real allowlist and broke for a second team member's tickets). These adapters target private repos/projects where label/triage access is already the trust boundary.
- Pipeline stage is now expressed purely via labels (`factory:ready-for-spec`, `factory:ready-to-implement`) instead of a GitHub Project Status field. Trigger `state` maps to the issue's native open/closed state. A human may still track progress on a project board for their own visualization; Factory never reads it.
- `.factory/workflows/triage/WORKFLOW.md` and `.factory/workflows/implement/WORKFLOW.md` updated so the agent removes the relevant label at the point that used to move the Project item, so the trigger doesn't refire on the same issue. Project-board moves are now explicitly optional/cosmetic.
- `factory init`'s generated config, `examples/config.toml`, this repo's own `.factory/config.toml`, `scripts/create-demo-issue.sh`, README.md, docs/design.md, and docs/local-v1.md updated to the label-only model.

## Operational note

This repo's live issues currently use the old single `factory:ready` label. Any issue relying on that label (paired with the old Project Status field) will stop matching once this merges — they need `factory:ready-for-spec` or `factory:ready-to-implement` instead.

## Test plan

- [x] `cargo build --locked`
- [x] `cargo test --locked --all-targets` (all suites pass, including new `tests/github_source_adapter.rs`)
- [x] `cargo fmt --all --check`
- [x] `cargo clippy --locked --all-targets -- -D warnings`
- [x] Ran `.factory/sources/github` directly against the real `gh` CLI and this repo's real issues — confirmed correct JSON output with no `author` field and no Projects/GraphQL calls
- [x] `factory init`, `factory validate`, `factory workflows` run clean against the updated config from a plain (non-worktree) clone
- [ ] Independent review in progress

Closes #93

### CODE REVIEW

File: .factory/sources/github [hunk 1]
```diff
@@ -32,63 +23,23 @@ while (($# > 0)); do
...
-          state: env.FACTORY_SOURCE_STATE,
-          labels: [.labels.nodes[].name],
-          url: .url,
-          author: .author.login
-        }
-    ' >"$results"
-then
-  exit 1
-fi
+args=(issue list --state "$state" --limit 1000 --json number,title,body,url,labels)
```
[the remark this answers is not in the document]
Comment (owainlewis) [reply .factory/sources/github]: Fixed in 764303d. `.factory/sources/github` now requests `max_results + 1` (1001) issues and fails loudly if the result meets or exceeds that count, instead of silently truncating at 1000, mirroring the existing Jira adapter's truncation check. Added tests covering both the truncation-failure case and the exact-boundary success case (`tests/github_source_adapter.rs`).

While adding those tests I also found and fixed a related latent bug: iterating an empty `labels` bash array under `set -u` throws "unbound variable" on macOS's default bash when a trigger is configured with zero labels — guarded that loop too.
